src/panels/config/helpers/forms/ha-counter-form.ts

```diff
@@ -155,14 +173,20 @@ class HaCounterForm extends LitElement {
     ev.stopPropagation();
     const target = ev.target as any;
     const configValue = target.configValue;
-    const value =
-      target.type === "number"
-        ? target.value !== ""
-          ? Number(target.value)
-          : undefined
-        : target.localName === "ha-switch"
-          ? (ev.target as HaSwitch).checked
-          : ev.detail?.value || target.value;
+    let value = ev.detail?.value || target.value;
+
+    if (target.type === "number") {
+      if (target.value !== "") {
+        value = Number(target.value);
+      } else {
+        value = undefined;
+      }
+    } else if (target.localName === "ha-switch") {
+      value = (ev.target as HaSwitch).checked;
+    } else if (target.localName === "ha-selector-boolean") {
+      value = ev.detail.value;
```

Comment:
```suggestion
```
This if seems redundant

Reply:
I see. So the issue is that we have `ev.detail?.value || target.value`. It should be `??` instead and then we shouldn't need this `if`. I added another change suggestion. Please try it out